### PR TITLE
implement get_tips response serialization

### PR DESCRIPTION
**Pull request by howjmay**
implement get_tips response serialization

implement hash243_stack_to_json_array()

### CODE REVIEW

**Comment by thibault-martinez** [comment]
Hi @HowJMay Thanks again for the PR !
There seems to be memory leaks, could you have a look at the CI ?

**Comment by howjmay** [comment]
Sor my fault ! 
I'm gonna fix it now

File: cclient/serialization/json/helpers.c [hunk 1]
```diff
@@ -261,7 +261,40 @@ retcode_t hash243_queue_to_json_array(hash243_queue_t queue, cJSON* const json_r
...
+
+    for (int i = array_count - 1; i >= 0; i--) {
+      tryte_t trytes_out[NUM_TRYTES_HASH + 1];
+      size_t trits_count = flex_trits_to_trytes(trytes_out, NUM_TRYTES_HASH,
+                                                hash243_stack_at(stack, i),
+                                                NUM_TRITS_HASH, NUM_TRITS_HASH);
+      trytes_out[NUM_TRYTES_HASH] = '\0';
+      if (trits_count != 0) {
+        cJSON_AddItemToArray(array_obj,
+                             cJSON_CreateString((const char*)trytes_out));
```
Comment (oopsmonk) [inline cclient/serialization/json/helpers.c]: `char const*`

File: cclient/serialization/json/tests/get_tips.c [hunk 2]
```diff
@@ -21,10 +21,45 @@ void test_serialize_get_tips(void) {
...
+      "\"" TEST_81_TRYTES_1
+      "\","
+      "\"" TEST_81_TRYTES_2
+      "\","
+      "\"" TEST_81_TRYTES_3 "\"]}";
+  get_tips_res_t* res = get_tips_res_new();
+
+  flex_trit_t trits_243[FLEX_TRIT_SIZE_243];
+  TEST_ASSERT(flex_trits_from_trytes(trits_243, NUM_TRITS_HASH,
+                                     (const tryte_t*)TEST_81_TRYTES_1,
```
Comment (oopsmonk) [inline cclient/serialization/json/tests/get_tips.c]: ditto

File: cclient/serialization/json/tests/get_tips.c [hunk 3]
```diff
@@ -21,10 +21,45 @@ void test_serialize_get_tips(void) {
...
+                                     NUM_TRYTES_HASH, NUM_TRYTES_HASH));
+  TEST_ASSERT(get_tips_res_hashes_add(res, trits_243) == RC_OK);
+  TEST_ASSERT(flex_trits_from_trytes(trits_243, NUM_TRITS_HASH,
+                                     (const tryte_t*)TEST_81_TRYTES_2,
```
Comment (oopsmonk) [inline cclient/serialization/json/tests/get_tips.c]: ditto

File: cclient/serialization/json/tests/get_tips.c [hunk 4]
```diff
@@ -21,10 +21,45 @@ void test_serialize_get_tips(void) {
...
+                                     NUM_TRYTES_HASH, NUM_TRYTES_HASH));
+  TEST_ASSERT(get_tips_res_hashes_add(res, trits_243) == RC_OK);
+  TEST_ASSERT(flex_trits_from_trytes(trits_243, NUM_TRITS_HASH,
+                                     (const tryte_t*)TEST_81_TRYTES_3,
```
Comment (oopsmonk) [inline cclient/serialization/json/tests/get_tips.c]: ditto

**Review by oopsmonk — CHANGES_REQUESTED** [review]

**Review by oopsmonk — APPROVED** [review]

**Pull request merged**